common/ferrostar/src/routing_adapters/osrm/mod.rs

```diff
@@ -87,6 +96,17 @@ impl Route {
                 } else {
                     WaypointKind::Break
                 },
+                properties: if waypoint.name.is_some() || waypoint.distance.is_some() {
+                    Some(
+                        serde_json::to_vec(&OsrmWaypointProperties {
+                            name: waypoint.name.clone(),
+                            distance: waypoint.distance,
+                        })
+                        .expect("Infallible JSON serialization"),
```

Comment:
This can't fail unless the data structure itself (`OsrmWaypointProperties`) does something that's illegal in JSON like non-string map keys, so this is reasonable to panic on.